apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,64 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
+  String variant
+
+  @InputFiles
+  Collection<File> getInputFiles() {
+    Set<File> inputFiles = Sets.newHashSet()
+    config.each { ext ->
+      ext.getFiles().asMap().entrySet().each { entry ->
+        entry.value.each { file ->
+          inputFiles.add(new File(entry.key, file))
+        }
+      }
+    }
+    return inputFiles
+  }
+
+  @Override
+  public void exec() {
+    File apolloScript = project.file(APOLLO_CODEGEN)
+    if (!apolloScript.isFile()) {
+      throw new GradleException("Apollo-codegen was not found in node_modules. Please run 'gradle " +
+          "${ApolloCodegenInstallTask.NAME}")
+    }
+
+    def schemaPath = getSchemaPath()
+    setScript(apolloScript)
+    List<String> apolloArgs = ["generate"]
+    apolloArgs.addAll(getInputFiles().collect { project.file(it).absolutePath })
```

Comment:
The way I see the flow would be:

ApolloCodgenTask would be dependent on ApolloIRgen Task. Then we register ApolloClassgenTask as a java generating task. 

```
This will make the generate[Variant]Sources task depend on this task and add the new source folders as compilation inputs. The new source folders are also added to the model.
```
So, yes it will be part of compilation definitely.